flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/InitTaskManagerDecorator.java

```diff
@@ -149,7 +153,16 @@ private Container decorateMainContainer(Container container) {
                 .withResources(resourceRequirements);
 
         // Merge fields
-        mainContainerBuilder.addAllToPorts(getContainerPorts()).addAllToEnv(getCustomizedEnvs());
+        mainContainerBuilder
+                .addAllToPorts(getContainerPorts())
+                .addAllToEnv(getCustomizedEnvs())
+                .addNewEnv()
+                .withName(ENV_FLINK_POD_NODE_ID)
+                .withValueFrom(
```

Comment:
I would like to suggest also setting the `ENV_FLINK_POD_NODE_ID` in standalone on K8s.

https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/resource-providers/standalone/kubernetes/

Reply:
I have concerns on this, because currently TMs in standalone clusters don't know the deployment environment (`standalone` or `standalone on k8s`). I prefer not do this, and maybe we can let the `TaskManagerOptionsInternal#TASK_MANAGER_NODE_ID` as public option later, and then users can configure it in yaml (like [config `taskmanager.resource-id`](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/resource-providers/standalone/kubernetes/#:~:text=args%3A%20%5B%22taskmanager%22%2C%20%22%2DDtaskmanager.resource%2Did%3D%24(POD_NAME)%22%5D))

Reply:
Even the standalone mode is not running on K8s, why we could not configure the node id?

Reply:
To do that, we need to introduce a new public option(`taskmanager.node-id`), it needs more discussions

Reply:
Then I am afraid the taskmanager location is not correct when the standalone cluster is deployed on the K8s. Because the hostname(aka external address) is the pod name, not the real node-id.

Not introducing a public config option `taskmanager.node-id` makes sense to me. Maybe we could make it configured via `ENV_FLINK_POD_NODE_ID`.

Reply:
Make sense.